### DISCUSSION TITLE
Question. Can I use Serverpod to make xmlrpc calls from the server?

### DISCUSSION THREAD

**Discussion by mdev88** [serverpod/serverpod#2448] [Q&A] [answered]
Hello, I am very interested in starting using Serverpod, but I'm not 100% sure it can do what I need...

I am developing a Flutter app that talks with an Odoo server using XMLRPC calls (for this I use the odoo_rpc package).

I'm wondering if I can use Serverpod (mini? web server?) and move the XMLRPC calls to the server, and then use it as a sort of proxy between my Flutter app and the Odoo server.

So I would, for example, send an API request to an endpoint from the Flutter app to Serverpod, then Serverpod would make the necessary XMLRPC calls to the Odoo server, and send a normal API response to the app with the information obtained from Odoo.

I hope I explained myself... :)

Thanks!

**Comment by SandPod** [comment, marked as answer]
Hey @mdev88 👋 

I converted this to questions so that others would more easily find it.

Yes, this should be simple to implement. You can use any dart package, just as usual, inside of the Serverpod Server.

Then you could leverage Serverpods RPC to communicate with the flutter app. Then you can simply proxy the requests inside of your server to your third party server.

The Serverpod server becomes sort of a [BFF](https://learn.microsoft.com/en-us/azure/architecture/patterns/backends-for-frontends) for your flutter application.
